wallet-new/src/Cardano/Wallet/API/WIP/Handlers.hs

```diff
@@ -39,14 +46,33 @@ deleteExternalWallet :: PassiveWalletLayer IO
 deleteExternalWallet _pwl _encodedRootPK =
     error "[CHW-106], Cardano Hardware Wallet feature, , delete external wallet, unimplemented yet."
 
+-- | Creates new unsigned transaction.
+--
+-- NOTE: This function does /not/ perform a payment, it just prepares raw
+-- transaction which will be signed and submitted to the blockchain later.
 newUnsignedTransaction :: ActiveWalletLayer IO
-                       -> PaymentWithChangeAddress
-                       -> Handler (WalletResponse RawTransaction)
-newUnsignedTransaction _awl _paymentWithChangeAddress =
-    error "[CHW-57], Cardano Hardware Wallet, unimplemented yet."
+                       -> Payment
+                       -> Handler (WalletResponse UnsignedTransaction)
+newUnsignedTransaction aw payment@Payment{..} = do
+    let inputGrouping = toInputGrouping $ fromMaybe (V1 defaultInputSelectionPolicy)
+                                                    pmtGroupingPolicy
+    res <- liftIO $ (WalletLayer.createUnsignedTx aw) inputGrouping
+                                                      SenderPaysFee
+                                                      payment
+    case res of
+        Left err -> throwM err
+        Right (tx, addrsAndPaths) -> do
+            let txInHexFormat = mkTransactionAsBase16 tx
+                srcAddrsWithDerivationPaths = NE.toList $
+                    NE.map (\(addr, path) -> AddressAndPath (mkAddressAsBase58 addr)
+                                                            (map word32ToAddressLevel path))
+                           addrsAndPaths
+                unsignedTx = UnsignedTransaction txInHexFormat
+                                                 srcAddrsWithDerivationPaths
+            return $ single unsignedTx
```

Comment:
If you desire, it's also possible to do this "v1 conversion" directly in the data layer, so that your handler becomes really a one liner. You can check the other handlers to get an idea. This has also the nice added bonus that now your `WalletLayer` API calls doesn't need to return the derivation paths, but returns a single nice `V1.UnsignedTransaction` 😉